src/components/ProductDetailsPage/ProductDetailsPage.js

```diff
@@ -1,85 +1,90 @@
 import { ADDED_TO_WISHLIST_NOTIFICATION } from '../../constants/notificationData';
-
 import './ProductDetailsPage.scss';
 
 export const CN = 'product-details';
+class ProductDetailsPage extends React.Component {
+  constructor(props) {
+    super(props);
 
-const ProductDetailsPage = (props) => {
-  const laptopWidth = 1024;
-  const [isVerticalCarousel, setVerticalCarousel] = useState(
-    window.innerWidth > laptopWidth
-  );
-  const {
-    match: {
-      params: { id }
-    },
-    history: { goBack },
-    products
-  } = props;
-
-  const product = products.find(({ _id }) => _id === id);
+    const laptopWidth = 1024;
 
-  window.onresize = () => setVerticalCarousel(window.innerWidth > laptopWidth);
-
-  if (products.length) {
-    const {
-      images, title, price, description
-    } = product;
+    this.state = {
+      isVerticalCarousel: window.innerWidth > laptopWidth
+    };
+  }
 
-    const productImageData = images.map(({ claudinaryId }) => ({
-      src: `${process.env.IMAGE_URL}/${claudinaryId}`
-    }));
+  setVerticalCarousel() {
+    this.state({
+      isVerticalCarousel: window.innerWidth > this.laptopWidth
+    });
+  }
 
-    const addToWishlistWithNotification = () => {
-      const { showMessage, addToWishlist } = props;
-      addToWishlist(id);
-      showMessage(ADDED_TO_WISHLIST_NOTIFICATION);
-    };
+  addToWishlistWithNotification = () => {
+    const { showMessage, addToWishlist, id } = this.props;
+    addToWishlist(id);
+    showMessage(ADDED_TO_WISHLIST_NOTIFICATION);
+  };
 
-    return (
-      <>
-        <Button className="go-back-btn" icon="arrow left" onClick={goBack}>
-          Go Back
-        </Button>
-        <div className={`${CN} content`}>
-          <ProductImage
-            className={`${CN}-images`}
-            images={productImageData}
-            verticalCarousel={isVerticalCarousel}
-          />
-          <div className={`${CN}-description`}>
-            <ProductOrder
-              id={id}
-              className={`${CN}-description__product-order`}
-              title={title}
-              price={`$${price}`}
-              description={description}
-              onClickAddToWishlist={addToWishlistWithNotification}
+  render() {
+    const {
+      match: {
+        params: { id }
+      },
+      history: { goBack },
+      products
+    } = this.props;
+    const product = products.find(({ _id }) => _id === id);
+    if (products.length) {
+      const {
+        images, title, price, description
+      } = product;
+      const productImageData = images.map(({ claudinaryId }) => ({
+        src: `${process.env.IMAGE_URL}/${claudinaryId}`
+      }));
+      const { isVerticalCarousel } = this.state;
+      return (
+        <>
+          <Button className="go-back-btn" icon="arrow left" onClick={goBack}>
+            Go Back
+          </Button>
+          <div className={`${CN} content`}>
+            <ProductImage
+              className={`${CN}-images`}
```

Comment:
Please change className={`${CN}-images`} to className={`${CN}__images`} (see my previous comment regarding BEM naming convention). Same for other classNames